ngraph/core/reference/include/ngraph/runtime/reference/matmul.hpp

```diff
@@ -38,6 +33,44 @@ namespace ngraph
     {
         namespace reference
         {
+            namespace details
+            {
+                template <typename T>
+                void dot(const T* arg0,
+                         const T* arg1,
+                         T* out,
+                         const Shape& arg0_shape,
+                         const Shape& arg1_shape,
+                         const Shape& out_shape)
+                {
+                    std::fill(out, out + shape_size(out_shape), static_cast<T>(0));
+                    size_t arg0_rank = arg0_shape.size();
```

Comment:
All of these variables should be const since they are not modified.

Reply:
Done